x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -2202,13 +2202,19 @@ private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
             return;
         }
 
-        final PlainActionFuture<ActionResponse.Empty> future = new UnsafePlainActionFuture<>(ThreadPool.Names.GENERIC);
-        getClient().execute(
-            ActionTypes.RELOAD_REMOTE_CLUSTER_CREDENTIALS_ACTION,
-            new TransportReloadRemoteClusterCredentialsAction.Request(settingsWithKeystore),
-            future
-        );
-        future.actionGet();
+        // Run this action in system context -- it was authorized upstream and should not be tied to end-user permissions
+        final ThreadContext ctx = getClient().threadPool().getThreadContext();
+        assert ctx != null : "Thread context must be set for reload call";
+        try (ThreadContext.StoredContext ignore = ctx.stashContext()) {
```

Comment:
Can we stash the context and mark as system context from `TransportNodesReloadSecureSettingsAction` instead of here ? It feels off that we can call this method directly from anywhere that we can reference  Security#reload and the work will be done in the system context, possibly having never run any authz. 

Since the expectation is that work is implicitly allowed as a sub action from  `TransportNodesReloadSecureSettingsAction`, I think having `TransportNodesReloadSecureSettingsAction` mark it as the system context is more correct since. It would also apply to all other sub reload actions, but that I think is also more correct since the permissions to use `TransportNodesReloadSecureSettingsAction` need to converge to a single permission.

Alternatively, I think we could change `RELOAD_REMOTE_CLUSTER_CREDENTIALS_ACTION` to an actual "internal:" action, helping to re-inforce that this is sub action / implementation detail of another action. But that would I prefer keeping the action name as-is and ensuring that the flip to system context is a bit more isolated to it's usage.

Reply:
I included it here primarily because it's an implementation detail that it's an action to begin with -- most other reload calls happen at the "service" layer and are not subject to authz at all -- it would be the same here if we had access to the right services within the Security plugin (we'd just call `RemoteClusterService.reload(...)` without the action workaround, but unfortunately, `RemoteClusterService` is not available in Security. To me it actually feels _more_ isolated to have it here.

However, I don't feel strongly -- `TransportNodesReloadSecureSettingsAction` is a fine place to switch to system context (though we'll have to double-check that we're not somehow switching back out of it along the way) -- let me know if you strongly prefer `TransportNodesReloadSecureSettingsAction` instead and I pull the switch over there.

Reply:
Discussed on Slack -- switching to system context in `TransportNodesReloadSecureSettingsAction` has several downsides. 

Since `Security#reload` is a public method that could (in theory) be called in a context that did not ensure proper authz it's however better to also avoid the system context switch inside reload.

To accommodate this, we've decided to simply rename the action to have a prefix that's covered by `manage`. Since it's a local-only action there are not BWC concerns with this.